File: src/main/Java/boj_step16/boj10773.java
import java.io.*;
import java.util.Stack;

public class boj10773 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(System.out));
        int k = Integer.parseInt(reader.readLine());
        Stack<Integer> stack = new Stack<>();
        while (k-- > 0) {
            int n = Integer.parseInt(reader.readLine());
            if (n == 0) stack.pop();
            else stack.push(n);
        }
        if (stack.isEmpty()) writer.append("0").close();
        else writer.append(String.valueOf(stack.stream().reduce(Integer::sum).get())).close();
    }
}

File: src/main/kotlin/boj_step16/boj10773.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*

fun main() {
    val r = BufferedReader(InputStreamReader(System.`in`))
    val w = BufferedWriter(OutputStreamWriter(System.out))
    val k = r.readLine().toInt()
    val stack = Stack<Int>()
    var sumMoney = 0
    repeat(k) {
        val money = r.readLine().toInt()
        if (money == 0) {
            sumMoney -= stack.pop()
        } else {
            stack.add(money)
            sumMoney += money
        }
    }
    w.write(sumMoney.toString())
    w.close()
}